test/integration/protocol_integration_test.cc

```diff
@@ -3163,6 +3208,25 @@ TEST_P(ProtocolIntegrationTest, UpstreamDisconnectBeforeResponseCompleteWireByte
   expectUpstreamWireBytesSentAndReceived(access_log_name_, 0, 159, 47, 0, 0, 113, 13, 0, 0);
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, BadRequest) {
+  // we only care about upstream protocol.
+  if (downstreamProtocol() != Http::CodecType::HTTP1) {
+    return;
+  }
+  useAccessLog("%DOWNSTREAM_WIRE_BYTES_SENT% %DOWNSTREAM_WIRE_BYTES_RECEIVED% "
+               "%DOWNSTREAM_BODY_BYTES_SENT% %DOWNSTREAM_BODY_BYTES_RECEIVED%\n");
+  initialize();
+  std::string response;
+  std::string full_request("\r\n", 100);
```

Comment:
Note: to create a string in C++ repeating a single character 100x you do:
`std::string my_str('\r', 100)`

What you're doing is invoking another string ctor (see: https://en.cppreference.com/w/cpp/string/basic_string/basic_string)